plugins/Goals/API.php

```diff
@@ -476,7 +476,16 @@ public function get($idSite, $period, $date, $segment = false, $idGoal = false,
         );
 
         foreach ($segments as $appendToMetricName => $predefinedSegment) {
-            $segmentToUse = $this->appendSegment($predefinedSegment, $segment);
+            if (!empty($predefinedSegment)) {
+                // we are disabling the archiving of these segments as the archiver archives them already using
+                // archiveProcessDependend logic. Otherwise we would eg archive reports that we don't need:
+                // userid=5;visitorType%3D%3Dnew;visitorType%3D%3Dreturning%2CvisitorType%3D%3DreturningCustomer
+                // userid=5;visitorType%3D%3Dreturning%2CvisitorType%3D%3DreturningCustomer;visitorType%3D%3Dnew;
+                // it would also archive dependends for these segments that we already combined here and then combine
+                // segments again when archiving dependends
+                Archiver::$ARCHIVE_DEPENDENT = false;
+            }
+            $segmentToUse = $this->appendSegment($segment, $predefinedSegment);
```

Comment:
flipped `segment` and `predefinedSegment` parameter.

Previously, in API we were using `$predefinedSegment, $segment` and then in the archiver we would have used `$segment, $predefinedSegment` meaning they actually resulted in different segment strings and resulted in quite a few additional segment archives that weren't even needed.